middleware/auth.js

```diff
@@ -26,22 +26,29 @@ exports.protect = asyncHandler(async (req, res, next) => {
 
     console.log(decoded);
 
-    req.user = await SuperAdmin.findById(decoded.id);
+    req.superadmin = await SuperAdmin.findById(decoded.id);
+    req.admin = await Admin.findById(decoded.id);
+    req.tutor = await Tutor.findById(decoded.id);
+    req.student = await Student.findById(decoded.id);
 
     next();
   } catch (err) {
     return next(new ErrorResponse("Not authorize to access this routes", 401));
-  } 
+  }
 });
 
 //Grant access to specific roles
 exports.authorize = (...roles) => {
   return (req, res, next) => {
-    if(!roles.includes(req.user.role)){
-      return next(new ErrorResponse(`Role ${req.user.role} is not authorized to access this route`,403));
+  
+    if (!roles.includes(req.user.role)) {
```

Comment:
Since you have dclared `superadmin`, `admin`, `tutor`, and `student` declared on the `req`  object in your `protect` function. Then you cannot use `req.user` here. Since `user` was not declared on the `req` object. I'm sure this didn't work when you tested the routes you used `authorize` for